templates/leptos-ssr/content/src/server.rs

```diff
@@ -1,19 +1,43 @@
-use leptos_spin::{render_best_match_to_stream, RouteTable, server_fn::register_explicit};
-use spin_sdk::http::{IncomingRequest, ResponseOutparam};
-use spin_sdk::http_component;
+use std::sync::Arc;
 
-#[http_component]
-async fn handle_{{project-name | snake_case}}(req: IncomingRequest, resp_out: ResponseOutparam) {
-    let mut conf = leptos::get_configuration(None).await.unwrap();
-    conf.leptos_options.output_name = "{{project-name | snake_case}}".to_owned();
+use leptos::config::get_configuration;
+use leptos_wasi::prelude::{Executor, IncomingRequest, ResponseOutparam, WasiExecutor};
+use wasi::exports::http::incoming_handler::Guest;
+use wasi::http::proxy::export;
 
-    // Register server functions
-    register_explicit::<crate::app::SaveCount>();
+use crate::app::{shell, App, SaveCount};
 
-    let app = crate::app::App;
+struct LeptosServer;
 
-    let mut routes = RouteTable::build(app);
-    routes.add_server_fn_prefix("/api").unwrap();
+impl Guest for LeptosServer {
```

Comment:
I wonder if we should use the `#[http_component]` macro (which allows the IncomingRequest/ResponseOutparam signature) instead of the raw binding here?  It may be more familiar to Spin users.  But maybe app developers don't touch this file enough to make any odds.

Reply:
I ran into some challenges with reconciling the types before when I dug into making that work -- I hit a dead-end trying to convert a Spin IncomingRequest into the one expected by leptos_wasi iirc. I will take another crack at this

Reply:
@ogghead Oh, if it doesn't work automatically then don't worry about it.  It's no biggie.

Reply:
Yeah, still running into challenges around converting types sadly -- I had gone down the route of newtypes to satisfy orphan rule before but it was becoming a rabbithole. This would be ideal though, I can create an issue to track that effort if you're on board with that

Reply:
Yeah it is definitely not worth investing more time in.  Again, sorry for sending you in an unsuccessful direction.

Reply:
Totally no worries! I spent more time than I would care to admit trying to get that working a couple weeks ago XD I do appreciate the feedback, as I did want to test whether ~Rohan would come to our aid~ Spin 3 or a new version of leptos_wasi might get this rolling